caas/kubernetes/provider/application.go

```diff
@@ -22,3 +23,13 @@ func (k *kubernetesClient) Application(name string, deploymentType caas.Deployme
 		k.randomPrefix,
 	)
 }
+
+// DesiredReplicas returns the desired replicas for the given application.
+func (k *kubernetesClient) DesiredReplicas(name string, deploymentType k8s.WorkloadType) (int, error) {
+	app := k.Application(name, k8s.WorkloadTypeStatefulSet)
```

Comment:
Don't use the core `WorkloadType` directly. This can cause issues if the core type is changed and we wonder why the application on k8s stops working.

Instead, switch on the core `WorkloadType` to give you a k8s one.

Reply:
As discussed offline, fixed this by simply removing the unused argument in `DesiredReplicas(name string)`.